platform/ios/src/MGLMapView.mm

```diff
@@ -5889,7 +5903,7 @@ - (void)didUpdateLocationSignificantlyAnimated:(BOOL)animated completionHandler:
      completionHandler:^{
         MGLMapView *strongSelf = weakSelf;
         if (strongSelf.userTrackingState == MGLUserTrackingStateBegan ||
-            strongSelf.userTrackingState == MGLDistanceThresholdForCameraPause)
+            strongSelf.userTrackingState == MGLUserTrackingStateBeginSignificantTransition)
```

Comment:
nice catch 👍 , and what a bug :D

Reply:
this bug is still in place in master of mapbox repo :)